### PR TITLE
Port Localization and ACL related code

### PR DESCRIPTION
**Pull request by wintondeshong**
Ported core logic for handling localization (via json culture files)
Ported core actors of ACL handing (more to come in form of conductors/extensions in other packages)
Ported some basic enumerations along the way as well

This fixes issue #6

### CODE REVIEW

**Comment by wintondeshong** [comment]
Will be creating issues to increase test coverage on added actors that didn't have any to begin with.

File: src/AndcultureCode.CSharp.Core/Constants/ContentTypes.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+namespace AndcultureCode.CSharp.Core.Constants
+{
+    public static class ContentTypes
+    {
+        public const string CSS = "text/css";
+        public const string HTML = "text/html";
+        public const string JAVASCRIPT = "text/javascript";
+        public const string JSON = "application/json";
+        public const string PDF = "application/pdf";
+        public const string XML = "application/xml";
```
Comment (nhinhdao) [inline src/AndcultureCode.CSharp.Core/Constants/ContentTypes.cs:10]: Align = (please ignore if it doesn't follow alignment standard like in Edio)

File: src/AndcultureCode.CSharp.Core/Extensions/ICultureExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+namespace AndcultureCode.CSharp.Core.Extensions
+{
+    public static class ICultureExtensions
+    {
+        #region Default
+
+        public static ICulture Default(this IEnumerable<ICulture> cultures)
+        {
+            var matches = cultures?.Where(e => e.IsDefault).ToList();
+            var matchesCount = matches == null ? 0 : matches.Count();
```
Comment (nhinhdao) [inline src/AndcultureCode.CSharp.Core/Extensions/ICultureExtensions.cs:18]: Can be shortened as `matches?.Count() ?? 0;`

File: src/AndcultureCode.CSharp.Core/Models/Localization/Culture.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        public List<CultureTranslation> CultureTranslations
+        {
+            get
+            {
+                if (_translations != null)
+                {
+                    return _translations;
+                }
+
+                _translations = new List<CultureTranslation>();
```
Comment (nhinhdao) [inline src/AndcultureCode.CSharp.Core/Models/Localization/Culture.cs:50]: Can be shortened as `_translations = _translations ?? new List<CultureTranslation>();`

**Pull request merged**